xdsl/pattern_rewriter.py

```diff
@@ -650,7 +660,7 @@ class PatternRewriteWalker:
     The walker will also walk recursively on the created operations.
     """
 
-    pattern: RewritePattern
+    pattern: RewritePattern = field(default_factory=NoRewrite)
```

Comment:
Why this change?

Reply:
It made the toy example a bit neater, but it could be dropped

Reply:
My gut says drop it FWIW